app/client/jsx/Room.jsx

```diff
@@ -1,275 +1,314 @@
-    * by creating a new component for that room and updating the getRoomType() mapping.
-    */
-    constructor(props) {
-        super(props)
-
-        this.useLocalSessions = Config.useLocalSessions || false
-
-        // These are the room types for which we show the map button
-        this.roomTypesWithMap = {
-            JITSI: true,
-            IFRAME: true,
-            CONTENT: true,
-            IMAGEMAP: true
-        }
-
-        // These are the room types which require doors
-        this.roomTypesWithDoors = {
-            JITSI: true,
-            IFRAME: true,
-            IMAGEMAP: true
-        }
-
-        this.state = {
-            room: this.props.currentRoom.room,
-            entered: false,
-            users: []
-        }
-
-        this.socketApi = new WebSocketApi()
-        this.socketApi.startPinging(this.props.user.id)
-        this.socketApi.on('user-event', this.props.updateUsers.bind(this))
-        this.socketApi.on(`poke-${this.props.user.id}`, this.handlePoke.bind(this))
-
-        this.computePokeUnlockedState()
+     * There are currently 3 different types of rooms:
+     *   1. Regular Jitsi room that just has video
+     *   2. Art room which has a small video panel and an image
+     *   3. Text-based adventure rooms where you have to make a decision
+     *   4. Image map based room to display an organized & navigable set of content
+     *   5. Special purpose rooms that exist at a different route
+     */
+    const roomData = this.getRoomData();
+    const jitsiData = {
+      displayName: this.props.user.username,
+      avatar: this.props.user.avatar,
+      roomName: this.state.room,
+      muteRoom: roomData.muteRoom,
+      hideVideo: roomData.hideVideo,
+      disableSharing: roomData.type === "IFRAME",
+    };
+    const combo = { ...roomData, ...jitsiData };
+    return {
+      ART: <ArtRoom jitsiData={jitsiData} art={roomData.art}></ArtRoom>,
+      JITSI: <JitsiVideo jitsiData={jitsiData}></JitsiVideo>,
+      IFRAME: (
+        <IFrameRoom
+          jitsiData={jitsiData}
+          iframeOptions={roomData.iframeOptions}
+        ></IFrameRoom>
+      ),
+      CHATSTREAM: <ChatStreamRoom {...combo}></ChatStreamRoom>,
+      ADVENTURE: (
+        <Adventure
+          options={roomData}
+          onClick={this.onAdventureClick.bind(this)}
+        ></Adventure>
+      ),
+      IMAGEMAP: (
+        <ImageMapRoom imageMapOptions={roomData.imageMapOptions}></ImageMapRoom>
+      ),
+    }[roomData.type];
+  }
+
+  getRoomDescription() {
+    if (this.props.rooms[this.state.room].description) {
+      return (
+        <div className="room-content">
+          <ReactMarkdown
+            source={this.props.rooms[this.state.room].description}
+            allowedTypes={[
+              "text",
+              "break",
+              "emphasis",
+              "strong",
+              "delete",
+              "link",
```

Comment:
Nice! This is great. I test it out just to see and it looks good: 
![Screen Shot 2021-03-25 at 5 31 34 PM](https://user-images.githubusercontent.com/4405597/112546585-fa0e1a00-8d8f-11eb-91ff-c8fee9aebedc.png)